bin/merge.py

```diff
@@ -3,43 +3,140 @@
 
 import json
 from sys import argv
+from pathlib import Path
 
 
-def merge(filenames, filecount):
-    merged = {
-        "symbols": {},
-        "bigrams": {},
-        "trigrams": {},
-    }
-
-    # merge dictionaries
-    for filename in filenames:
-        with open(filename, "r") as corpus:
-            data = json.load(corpus)
-            for section in merged.keys():
-                for key, count in data[section].items():
-                    if key not in merged[section]:
-                        merged[section][key] = 0.0
-                    merged[section][key] += count / filecount
-
-    # sort the merged dictionary by symbol frequency (requires CPython 3.6+)
-    def sort_by_frequency(table, precision=2):
-        sorted_dict = {}
-        for key, count in sorted(table.items(), key=lambda x: -x[1]):
-            freq = round(count, precision)
-            if freq > 0:
-                sorted_dict[key] = freq
-        return sorted_dict
-
-    results = {}
-    results["corpus"] = ""
-    results["symbols"] = sort_by_frequency(merged["symbols"])
-    results["bigrams"] = sort_by_frequency(merged["bigrams"], 4)
-    results["trigrams"] = sort_by_frequency(merged["trigrams"])
-    return results
+# sort the merged dictionary by symbol frequency (requires CPython 3.6+)
+def _sort_ngram_by_frequency(table, precision=3):
+    sorted_dict = {}
+    for key, count in sorted(table.items(), key=lambda x: -x[1]):
+        freq = round(count, precision)
+        if freq > 0:
+            sorted_dict[key] = freq
+    return sorted_dict
+
+
+def sort_by_frequency(corpus: dict, precision=3):
+    for ngram in range(1, len(corpus["freq"].keys())):
+        ngram = str(ngram)
+        corpus["freq"][ngram] = _sort_ngram_by_frequency(
+            corpus["freq"][ngram], precision
+        )
+    return corpus
+
+
+def merge(filenames, output_name: str, mergetype: str, arg: dict = {}) -> dict:
+    """merge liste of JSON files output from `chardict` to get a sigle corpus file
+    the merge is on the number of character in each file"""
+
+    def _merge_dicts(corpus1: dict, corpus2: dict, mergetype: str, arg: dict) -> dict:
```

Comment:
I’m worried by the fact this only takes two corpora as inputs. For our use case we really need to mix N corpora with the same “weight” (to compute average frequencies).

Reply:
This is why I introduced a `filecount` variable to assign the weight of each corpus. 
For instance, in the current implementation ; corpuses are merge one by one, by the with a weight of 
`filecount/(filecount+1)`

Reply:
This might be easier for you but for most people it’s the opposite: calculating a sum or an average on all items in one go is more intuitive — and much sharper, as it avoids a succession of rounding approximations.